### PR TITLE
tutorial for Go implicit grant login

### PR DESCRIPTION
**Pull request by michaeljroddy**

### CODE REVIEW

File: oauth-implicit/go/index.html [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <title>Implicit Login</title>
+        <script src="https://sdk-cdn.mypurecloud.com/external/go-cli/axios/0.23.0/axios.min.js"></script>
```
Comment (timsmithgenesys) [inline oauth-implicit/go/index.html:5]: We should use a public CDN here. We don't really intend for people to consume axios from our CDN directly in their custom apps, and we don't provide any documentation around our rehosting of it.

File: oauth-implicit/go/main.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+)
+
+func main() {
+	openBrowser("http://localhost:5000/")
```
Comment (ronanwatkins) [inline oauth-implicit/go/main.go:16]: You should create variables for the host name and port like with the python tutorial

Comment (ronanwatkins) [reply oauth-implicit/go/main.go:16]: and environment

File: oauth-implicit/go/main.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+			result, err := validateToken(token)
+			if err != nil {
+				log.Fatal(err)
+			}
+			fmt.Println(result)
+		}
+	})
+
+	http.HandleFunc("/error/", func(writer http.ResponseWriter, request *http.Request) {
+		writer.Header().Set("Content-Type", "text/html; charset=utf-8")
```
Comment (ronanwatkins) [inline oauth-implicit/go/main.go:37]: Should use writer.WriteHeader to send a http.StatusNotFound code here as well

File: oauth-implicit/go/main.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+		fmt.Fprint(writer, "<html><p> 404. Not found. </p></html>")
+	})
+
+	err := http.ListenAndServe(":5000", nil)
```
Comment (ronanwatkins) [inline oauth-implicit/go/main.go:41]: Use the port variable for this when it's created

File: oauth-implicit/go/main.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// validates the access token returned from the implicit grant login
+func validateToken(token string) (string, error) {
+	environment := os.Getenv("GENESYS_CLOUD_ENVIRONMENT")
+
+	client := &http.Client{Timeout: time.Duration(2) * time.Second}
```
Comment (ronanwatkins) [inline oauth-implicit/go/main.go:51]: You can increase this timeout a little bit. 10s would be fine

File: oauth-implicit/go/main.go [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+	// create a new request for GET api/v2/users/me
+	req, err := http.NewRequest("GET", fmt.Sprintf("https://api.%s/api/v2/users/me", environment), nil)
```
Comment (ronanwatkins) [inline oauth-implicit/go/main.go:54]: http.MethodGet instead of "GET"

Comment (ronanwatkins) [inline oauth-implicit/go/main.go:54]: http.MethodGet instead of "GET"

**Review by ronanwatkins — APPROVED** [review]

**Pull request merged**